pkg/image/api/helper.go

```diff
@@ -184,7 +184,41 @@ func ParseDockerImageReference(spec string) (DockerImageReference, error) {
 		ref.ID = id
 		break
 	default:
-		// TODO: this is no longer true with V2
+		// Handle multiple segments in form: registry/namespace/namesegment1/namesegment2/.../name
+		// but not this form: http://registry/namespace/name. We don't support using the schema in
+		// the DockerImageReference.
+		if len(repoParts) > 3 && !strings.HasSuffix(repoParts[0], ":") {
+			if IsRegistryDockerHub(repoParts[0]) {
+				// DockerHub doesn't support multiple segments.
+				return ref, fmt.Errorf("the docker pull spec %q must be two or three segments separated by slashes", spec)
+			}
+			if len(repoParts[0]) == 0 {
+				return ref, fmt.Errorf("the docker pull spec %q must contain non-empty registry name", spec)
+			}
+			ref.Registry = repoParts[0]
+
+			if len(repoParts[1]) == 0 {
+				return ref, fmt.Errorf("the docker pull spec %q must contain non-empty namespace", spec)
+			}
+			ref.Namespace = repoParts[1]
+
+			for i, part := range repoParts[2:] {
+				if len(part) == 0 {
+					continue
+				}
+				if i > 0 {
+					ref.Name += "/"
+				}
+				ref.Name += part
+			}
+			if len(ref.Name) == 0 {
+				return ref, fmt.Errorf("the docker pull spec %q must be three or more segments separated by slashes", spec)
```

Comment:
@ncdc Actually this `if` is useless because if have more than 3 parts in slice (`repoParts`), two of them are not empty (`Registry` and `Namespace`) and rest of them are not empty as well after check inside `for`. It means that `Name` can't be empty.
I will remove this check.